core/src/apps/homescreen/__init__.py

```diff
@@ -67,18 +67,19 @@ def get_features() -> Features:
             Capability.U2F,
             Capability.Shamir,
             Capability.ShamirGroups,
+            Capability.PassphraseEntry,
         ]
     f.sd_card_present = io.SDCard().present()
     f.sd_protection = storage.sd_salt.is_enabled()
     f.wipe_code_protection = config.has_wipe_code()
+    f.session_id = cache.get_session_id()
+    f.passphrase_always_on_device = storage.device.get_passphrase_always_on_device()
     return f
 
 
 async def handle_Initialize(ctx: wire.Context, msg: Initialize) -> Features:
-    if msg.state is None or msg.state != cache.get_state(prev_state=bytes(msg.state)):
+    if msg.session_id is None or msg.session_id != cache.get_session_id():
```

Comment:
if the client sends session id, and it's different from the current session, then they get a new random session

I'm pretty sure that this is what we want, but wanted to ask anyway: are there reasons for/against using the client-provided session id?

Reply:
one comes to mind: host can tell whether or not their session was active on the device

Reply:
You mean to return the same session id as provided by the host even though it was not cached?

That is problematic, consider this behaviour:

- Host sends Initialize(), receives Features(session_id=A) with session id A.
- Someone else sends Initialize(), receives Features(session_id=B) and stuff happens.
- Host sends Initialize(session_id=A) which they stored and would receive Features(session_id=A), but the cache was cleared and Trezor needs to ask for passphrase again, which is wrong! The Host assumes that if the session id from Trezor is the same as the one they sent that it is the same session.

Does it make sense?

Reply:
well, my question was mostly "do we actually want this behavior?"
what you described is a usecase where we do want it. so i suppose that's a yes

Reply:
Yes :)